trio/_highlevel_open_tcp_listeners.py

```diff
@@ -150,14 +155,14 @@ async def open_tcp_listeners(port, *, host=None, backlog=None):
 
 
 async def serve_tcp(
-    handler,
-    port,
+    handler: Callable[[trio.SocketStream], Awaitable[Any]],
+    port: int,
     *,
-    host=None,
-    backlog=None,
-    handler_nursery=None,
-    task_status=trio.TASK_STATUS_IGNORED,
-):
+    host: str | bytes | None = None,
+    backlog: int | None = None,
```

Comment:
For `backlog` here, we need to check usages - a lot of Trio functions accept int + `math.inf`, which does look like it'll work here. So this should be a `float` actually. Perhaps that should be deprecated in favour of `None`.

Reply:
I agree, accepting float would be confusing and would not represent what we are trying to do here

Reply:
deprecating inf->None should be done in a separate PR, as it's used widely across the codebase. Several other functions have been typed as accepting float, so if it makes sense here I agree it should be added - though you can add a check that if it's `float` it's `math.inf`

Reply:
Bump on this @CoolCat467

Reply:
So making sure, we are going to make it accept `float` here then, but only return `int`?

Reply:
I don't understand what you mean by "return" here -- I'm just talking about supporting `inf` at least in this PR and unsupporting it in another one!

Reply:
Fixed in e90e4cc